lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+        this.objectMapper = objectMapper;
+        this.configService = configService;
+        this.userIdentityService = userIdentityService;
+        this.verifiableCredentialService = verifiableCredentialService;
+        this.reportUserIdentityService = reportUserIdentityService;
+        this.vcStoreItemScanDynamoDataStore = vcStoreItemScanDynamoDataStore;
+        this.reportUserIdentityScanDynamoDataStore = reportUserIdentityScanDynamoDataStore;
+        this.reportSummaryScanDynamoDataStore = reportSummaryScanDynamoDataStore;
+    }
+
+    @SuppressWarnings({"unused", "java:S107"}) // Used by AWS
+    public ReportUserIdentityHandler() {
+        this.objectMapper = new ObjectMapper();
+        this.configService = ConfigService.create();
+        this.userIdentityService = new UserIdentityService(configService);
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
+        this.reportUserIdentityService = new ReportUserIdentityService();
+        this.vcStoreItemScanDynamoDataStore =
+                new ScanDynamoDataStore<>(
+                        configService.getEnvironmentVariable(
+                                EnvironmentVariable.USER_ISSUED_CREDENTIALS_TABLE_NAME),
+                        VcStoreItem.class,
+                        DynamoDataStore.getClient(),
+                        configService);
+        this.reportUserIdentityScanDynamoDataStore =
+                new ScanDynamoDataStore<>(
+                        configService.getEnvironmentVariable(
+                                EnvironmentVariable.REPORT_USER_IDENTITY_TABLE_NAME),
+                        ReportUserIdentityItem.class,
+                        DynamoDataStore.getClient(),
+                        configService);
+        this.reportSummaryScanDynamoDataStore =
+                new ScanDynamoDataStore<>(
+                        configService.getEnvironmentVariable(
+                                EnvironmentVariable.REPORT_SUMMARY_TABLE_NAME),
+                        ReportSummaryItem.class,
+                        DynamoDataStore.getClient(),
+                        configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
+            throws IOException {
+        LogHelper.attachComponentId(configService);
+        LOGGER.info(LogHelper.buildLogMessage("Start processing report."));
+
+        ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult =
+                ReportProcessingResult.builder();
+        try {
+            ReportProcessingRequest reportProcessingRequest =
+                    objectMapper.readValue(inputStream, ReportProcessingRequest.class);
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Request received- " + reportProcessingRequest.toString()));
```

Comment:
Will this request contain any PII (i.e. user ids) in the continuation token? I think we can probably just drop this log tbh.

Reply:
Done